homeassistant/components/shelly/config_flow.py

```diff
@@ -53,13 +53,15 @@ async def validate_input(hass: core.HomeAssistant, host, data):
     return {
         "title": device.settings["name"],
         "hostname": device.settings["device"]["hostname"],
+        "sleep_period": get_device_sleep_period(device.settings),
+        "model": device.settings["device"]["type"],
     }
 
 
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Shelly."""
 
-    VERSION = 1
+    VERSION = 2
```

Comment:
Migrations are really tough, as it prevents users from downgrading and should be avoided as much as possible. Try to just do it inside `async_setup_entry`.